src/hyperion/device_setup_plans/dcm_pitch_roll_mirror_adjuster.py

```diff
@@ -0,0 +1,167 @@
+import bluesky.plan_stubs as bps
+from dodal.devices.DCM import DCM
+from dodal.devices.hfm import HFM
+from dodal.devices.i0 import I0
+from dodal.devices.qbpm1 import QBPM1
+from dodal.devices.s4_slit_gaps import S4SlitGaps
+from dodal.devices.vfm import VFM
+from ophyd import EpicsMotor, EpicsSignalRO
+
+from hyperion.device_setup_plans.peak_finder import (
+    PeakFinder,
+    SimpleMaximumPeakEstimator,
+    SingleScanPassPeakFinder,
+)
+from hyperion.log import LOGGER
+from hyperion.utils.lookup_table import LinearInterpolationLUTConverter
+
+RESET_SLIT_GAP_GROUP = "reset_slit_gap"
+
+PREP_FOR_VFM_GROUP = "prepare_for_vfm_align"
+PREP_FOR_HFM_GROUP = "prepare_for_hfm_align"
+
+# Magic values from GDA beamLineSpecificEnergy.py
+SLIT_SIZE_WIDE_OPEN_MM = (500, 500)
+SLIT_SIZE_LARGE_MM = (500, 60)
+SLIT_SIZE_SMALL_MM = (100, 60)
+
+VFM_FINE_PITCH_INITIAL_PRESET_MM = 0.019
+VFM_FINE_PITCH_ADJUST_PEAK_WIDTH_MM = 0.012
+VFM_FINE_PITCH_ADJUST_STEP_MM = 0.0006
+
+HFM_FINE_PITCH_INITIAL_PRESET_MM = 0.017
+HFM_FINE_PITCH_ADJUST_PEAK_WIDTH_MM = 0.016
+HFM_FINE_PITCH_ADJUST_STEP_MM = 0.0006
+
+DCM_PITCH_ADJUST_PEAK_WIDTH = 0.075
+DCM_PITCH_ADJUST_PEAK_STEP = 0.002
+
+
+class DCMPitchRollMirrorAdjuster:
+    """Auto-adjust the DCM, then re-adjust pitch, roll and vertical + horizontal focusing mirrors."""
+
+    def __init__(
+        self,
+        dcm: DCM,
+        vfm: VFM,
+        hfm: HFM,
+        qbpm1: QBPM1,
+        collimation_slits: S4SlitGaps,
+        i0: I0,
+        peak_finder: PeakFinder = SingleScanPassPeakFinder(
+            SimpleMaximumPeakEstimator()
+        ),
+    ):
+        super().__init__()
+        self._i0 = i0
+        self._collimation_slits = collimation_slits
+        self._peak_finder = peak_finder
+        self._dcm = dcm
+        self._vfm = vfm
+        self._hfm = hfm
+        self._qbpm1 = qbpm1
+
+    def auto_adjust_dcm_hfm_vfm_pitch_roll(self):
+        """Main entry point
+        Args:
+        vfm: Vertical Focus Mirror device
+        """
+
+        # TODO set transmission 1.0 ???
+
+        # XXX where is this value currently initialised in GDA?
+        current_pitch = yield from bps.rd(self._dcm.pitch.user_readback)
+        LOGGER.info("DCM Auto Pitch adjustment")
+        yield from self._adjust_to_peak(
+            self._dcm.pitch,
+            self._qbpm1.intensityC,
+            current_pitch,
+            DCM_PITCH_ADJUST_PEAK_WIDTH,
+            DCM_PITCH_ADJUST_PEAK_STEP,
+        )
+
+        # XXX GDA waits here for 2 seconds, not sure what we are waiting for here?
+        yield from bps.sleep(2)
```

Comment:
Should: Our general pattern with these sleeps in GDA has been to remove them, work out what they're actually waiting on when we get to the beamline and then refactor to wait on it properly